From d5e7cd24859924bf0fed9fbed2c9159a9789be10 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 22 Jan 2019 12:11:18 +1100
Subject: [PATCH] comdlg32: Improve Printer Setup dialog

Reset the Tab order to be more like windows.
Added Help Button.
Enabled Network button + messsage on Click.


Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/comdlg32/comdlg32.rc | 21 +++++++++++----------
 dlls/comdlg32/printdlg.c  |  4 +++-
 2 files changed, 14 insertions(+), 11 deletions(-)

diff --git a/dlls/comdlg32/comdlg32.rc b/dlls/comdlg32/comdlg32.rc
index 04ce3da9..d96e017 100644
--- a/dlls/comdlg32/comdlg32.rc
+++ b/dlls/comdlg32/comdlg32.rc
@@ -391,22 +391,18 @@ STYLE DS_MODALFRAME | WS_POPUP | WS_VISIBLE | WS_CAPTION | WS_SYSMENU | DS_CONTE
 CAPTION "Print Setup"
 FONT 8, "MS Shell Dlg"
 BEGIN
-    DEFPUSHBUTTON   "OK",IDOK,176,156,50,14,WS_GROUP
-    PUSHBUTTON      "Cancel",IDCANCEL,230,156,50,14
-/*    PUSHBUTTON      "Network...", psh5, 284,156,48,14 */
-
     GROUPBOX        "Printer",        grp4,   8,  4, 272,84, WS_GROUP
-    PUSHBUTTON      "&Properties",    psh2, 212, 17,  60,14, WS_GROUP
     LTEXT           "&Name:",         stc6,  16, 20,  36,8
-    COMBOBOX                          cmb1,  52, 18, 152,152,CBS_DROPDOWNLIST | CBS_SORT | WS_VSCROLL | WS_GROUP | WS_TABSTOP
+    COMBOBOX                          cmb1,  60, 18, 152,152,CBS_DROPDOWNLIST | CBS_SORT | WS_VSCROLL | WS_GROUP | WS_TABSTOP
+    PUSHBUTTON      "&Properties",    psh2, 212, 17,  60,14, WS_GROUP
     LTEXT           "Status:",        stc8,  16, 36,  47,10, SS_NOPREFIX
-    LTEXT           "",               stc12, 65, 36, 212,10, SS_NOPREFIX | SS_LEFTNOWORDWRAP
+    LTEXT           "",               stc12, 60, 36, 212,10, SS_NOPREFIX | SS_LEFTNOWORDWRAP
     LTEXT           "Type:",          stc7,  16, 48,  47,10, SS_NOPREFIX
-    LTEXT           "",               stc11, 65, 48, 212,10, SS_NOPREFIX | SS_LEFTNOWORDWRAP
+    LTEXT           "",               stc11, 60, 48, 212,10, SS_NOPREFIX | SS_LEFTNOWORDWRAP
     LTEXT           "Where:",         stc10, 16, 60,  47,10, SS_NOPREFIX
-    LTEXT           "",               stc14, 65, 60, 212,10, SS_NOPREFIX | SS_LEFTNOWORDWRAP
+    LTEXT           "",               stc14, 60, 60, 212,10, SS_NOPREFIX | SS_LEFTNOWORDWRAP
     LTEXT           "Comment:",       stc9,  16, 72,  47,10, SS_NOPREFIX
-    LTEXT           "",               stc13, 65, 72, 212,10, SS_NOPREFIX | SS_LEFTNOWORDWRAP
+    LTEXT           "",               stc13, 60, 72, 212,10, SS_NOPREFIX | SS_LEFTNOWORDWRAP
 
     GROUPBOX        "Paper",          grp2,   8, 92, 164,56, WS_GROUP
     LTEXT           "Si&ze:",         stc2,  16,108,  36, 8
@@ -418,6 +414,11 @@ BEGIN
     ICON            "",               ico1, 195,112,  18,20, WS_GROUP
     CONTROL         "P&ortrait",      rad1,"Button",BS_AUTORADIOBUTTON | WS_GROUP |WS_TABSTOP,224,106,52,12
     CONTROL         "L&andscape",     rad2,"Button",BS_AUTORADIOBUTTON,224,126,52,12
+
+    DEFPUSHBUTTON   "OK",IDOK,176,156,50,14,WS_GROUP
+    PUSHBUTTON      "Cancel",IDCANCEL,230,156,50,14
+    PUSHBUTTON      "&Help",pshHelp,8,156,50,14, WS_TABSTOP
+    PUSHBUTTON      "Network...", psh5, 65,156,50,14, WS_TABSTOP
 END
 
 PAGESETUPDLGORD DIALOG 32, 32, 240, 240
diff --git a/dlls/comdlg32/printdlg.c b/dlls/comdlg32/printdlg.c
index 69e9aeb..9bf17e0 100644
--- a/dlls/comdlg32/printdlg.c
+++ b/dlls/comdlg32/printdlg.c
@@ -1917,7 +1917,9 @@ static LRESULT PRINTDLG_WMCommandW(HWND hDlg, WPARAM wParam,
 	 ClosePrinter(hPrinter);
          break;
        }
-
+    case psh5: /* Network */
+        MessageBoxA(hDlg, "Not implemented yet!", "Error", MB_OK | MB_ICONEXCLAMATION);
+        break;
     case rad1: /* Paperorientation */
         if (lppd->Flags & PD_PRINTSETUP)
         {
-- 
1.9.1

